src/generators/utils/pagingOperations.ts

```diff
@@ -0,0 +1,378 @@
+        operation.description
+      )
+    ]
+  });
+
+  // Extract the parameter names for the All method to call it
+  let allMethodParameters = pagingMethodSettings.allMethod.parameters
+    .map(p => p.name)
+    .join(",");
+
+  // Extract the parameter names for the page method to call it
+  const pageMethodNameParams = pagingMethodSettings.pageMethod.parameters
+    .map(p => p.name)
+    .join(",");
+
+  method.addStatements([
+    `const iter = this.${pagingMethodSettings.allMethod.name}(${allMethodParameters});`,
+    `return {
+        next() {
+          return iter.next();
+        },
+        [Symbol.asyncIterator]() {
+          return this;
+        },
+        byPage: () => {
+          return this.${pagingMethodSettings.pageMethod.name}(${pageMethodNameParams});
+        }
+      };`
+  ]);
+}
+
+/**
+ * Generates the All method which loops through all the pages and returns a single array with all the results
+ */
+function writeAllMethod(
+  operationGroupClass: ClassDeclaration,
+  pagingMethodSettings: PagingMethodSettings
+) {
+  const returnType = `AsyncIterableIterator<${pagingMethodSettings.bodyResponseType}>`;
+
+  // Gets the page method parameters to use when calling it.
+  const pageMethodParameters = pagingMethodSettings.pageMethod.parameters
+    .map(p => p.name)
+    .join(",");
+
+  const method = operationGroupClass.addMethod({
+    name: `*${pagingMethodSettings.allMethod.name}`,
+    parameters: pagingMethodSettings.initialMethod.parameters,
+    scope: Scope.Private,
+    returnType,
+    isAsync: true
+  });
+
+  method.addStatements([
+    `for await (const page of this.${pagingMethodSettings.pageMethod.name}(${pageMethodParameters})) {
+        yield *page
+     }`
+  ]);
+}
+
+function writePageMethod(
+  operation: OperationDetails,
+  operationGroupClass: ClassDeclaration,
+  pagingMethodSettings: PagingMethodSettings
+) {
+  const returnType = `AsyncIterableIterator<${pagingMethodSettings.bodyResponseType}[]>`;
+
+  // Name of the property that contains the page value
+  const itemName = pagingMethodSettings.itemName;
+
+  // Name of the property that contains the nextLink
+  const nextLinkProperty = normalizeName(
+    pagingMethodSettings.nextLinkName!,
+    NameType.Property
+  );
+
+  // Extract the names for the initial method parameters
+  const initialMethodParameters = pagingMethodSettings.initialMethod.parameters
+    .map(p => p.name)
+    .join(",");
+
+  const method = operationGroupClass.addMethod({
+    name: `*${pagingMethodSettings.pageMethod.name}`,
+    parameters: pagingMethodSettings.pageMethod.parameters,
+    scope: Scope.Private,
+    returnType,
+    isAsync: true
+  });
+
+  let firstRequestStatements = [
+    `let result = await this.${pagingMethodSettings.initialMethod.name}(${initialMethodParameters});`
+  ];
+
+  if (operation.isLRO) {
+    // Since this is also an LRO operation, we need to poll until done to get the result
```

Comment:
can you put an example for generated methods for a paginated LRO API in the PR description?

Reply:
Sure, added it to the description